radicle-httpd/src/lib.rs

```diff
@@ -95,8 +88,60 @@ pub async fn run(options: Options) -> anyhow::Result<()> {
         )
         .into_make_service_with_connect_info::<SocketAddr>();
 
-    axum::Server::bind(&options.listen)
+    axum::Server::bind(&listen)
         .serve(app)
         .await
         .map_err(anyhow::Error::from)
 }
+
+/// Create a router consisting of other sub-routers.
+fn router(options: Options, home: Home) -> anyhow::Result<Router> {
+    let profile = Arc::new(radicle::Profile::load_at(home)?);
+
+    tracing::info!("using radicle home at {}", profile.home().display());
+
+    let ctx = api::Context::new(profile.clone());
+
+    let api_router = api::router(ctx);
+    let git_router = git::router(profile.clone(), options.aliases);
+    let raw_router = raw::router(profile);
+
+    let app = Router::new()
+        .merge(git_router)
+        .nest("/api", api_router)
+        .nest("/raw", raw_router);
+
```

Comment:
also I tried to get it as close as possible to the production router, but it doesn't play nice as once you call
```rust
.into_make_service_with_connect_info(..)
```

you can no longer use this in test:
```rust
.layer(MockConnectInfo(SocketAddr::from(([0, 0, 0, 0], 8080))));
```